MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetSource.cs

```diff
@@ -5,26 +5,35 @@
 namespace MixedRealityExtension.Assets
 {
     /// <summary>
-    /// Documents the origin of an asset
+    /// Documents the origin of an asset.
+    /// NOTE: The InternalId is not used for equality testing.
     /// </summary>
-    public struct AssetSource
+    public class AssetSource
     {
         /// <summary>
         /// The type of container the asset came from.
         /// </summary>
-        public AssetContainerType ContainerType { get; }
+        public AssetContainerType ContainerType { get; set; }
 
         /// <summary>
         /// The URL of the asset's container.
         /// </summary>
-        public Uri Uri { get; }
+        public string Uri { get; set; }
+
+        private Uri parsedUri;
+        /// <summary>
+        /// The parsed URI of the asset's container.
+        /// </summary>
+        public Uri ParsedUri => parsedUri = parsedUri ?? new Uri(Uri);
 
         /// <summary>
         /// The location of the asset within the container. Type-dependent.
         /// </summary>
-        public string InternalId { get; }
+        public string InternalId { get; set; }
```

Comment:
>InternalId [](start = 22, length = 10)

Exclude this field from JSON serialization as well?

Reply:
No, this field is supported on both sides. It's unused right now, but I think we'll need it when I implement UnloadAsset.

---
In reply to: [251068885](https://github.com/Microsoft/mixed-reality-extension-unity/pull/15#discussion_r251068885) [](ancestors = 251068885)